File: bindings/c/src/check_group.cpp
// Copyright (c) 2025 Daniel T. McGinnis
// SPDX-License-Identifier: MIT

#include <nux/check_group.h>
#include "handles.hpp"

NUXC_EXTERNC
nux_check_group_t*
nux_check_group_create ()
{
    return new nux_check_group_t{nux::check_group<nux::widget>{}};
}

NUXC_EXTERNC
void
nux_check_group_destroy (
    nux_check_group_t *check_group
) {
    delete check_group;
}

File: bindings/rs/nux/src/check_group.rs
// Copyright (c) 2025 Daniel T. McGinnis
// SPDX-License-Identifier: MIT

use crate::bindings::*;

pub struct CheckGroup {
    pub(crate) native_handle: *mut nux_check_group_t
}

impl CheckGroup {
    pub fn new () -> Self {
        return Self {
            native_handle: unsafe {
                nux_check_group_create()
            }
        };
    }
}

impl Drop for CheckGroup {
    fn drop (
        &mut self
    ) {
        unsafe {
            nux_check_group_destroy(self.native_handle)
        };
    }
}
